modules/custom/openy_upgrade_tool/openy_upgrade_tool.install

```diff
@@ -5,45 +5,149 @@
-  $config .= 'views.view.openy_upgrade_dashboard.yml';
-  $params = [
-    'display.page_1.display_options.path',
-    'display.page_1.display_options.menu.parent',
-    'display.page_1.display_options.menu.title',
-  ];
-  foreach ($params as $param) {
-    $config_importer->update($config,
-      'views.view.openy_upgrade_dashboard',
-      $param
-    );
+  // views.view.openy_upgrade_dashboard will be updated in next hooks.
+}
+
+/**
+ * Install OpenyUpgradeLog entity.
+ */
+function openy_upgrade_tool_update_8003() {
+  $entity_type = \Drupal::entityTypeManager()->getDefinition('openy_upgrade_log');
+  \Drupal::service('entity.definition_update_manager')->installEntityType($entity_type);
+}
+
+/**
+ * Switch from Logger entity to OpenyUpgradeLog.
+ */
+function openy_upgrade_tool_update_8004(&$sandbox) {
+  // We can't run this in post_update because this must be executed before
+  // openy_upgrade_tool_update_8005 update.
+  if (!isset($sandbox['progress'])) {
+    $sandbox['progress'] = 0;
+    $sandbox['current'] = 0;
+    $sandbox['max'] = \Drupal::entityQuery('logger_entity')
+      ->condition('type', 'openy_config_upgrade_logs')
+      ->count()
+      ->execute();
+  }
+
+  // Migrate openy_config_upgrade_logs in chunks of 20 items.
+  $ids = \Drupal::entityQuery('logger_entity')
+    ->condition('type', 'openy_config_upgrade_logs')
+    ->condition('id', $sandbox['current'], '>')
+    ->range(0, 20)
+    ->sort('id')
+    ->execute();
+
+  $configFactory = \Drupal::configFactory();
+  $entityTypeManager = \Drupal::entityTypeManager();
+  $logger_entities = $entityTypeManager
+    ->getStorage('logger_entity')
+    ->loadMultiple($ids);
+  foreach ($logger_entities as $logger_entity) {
+    $config_name = $logger_entity->getName();
+    $new_upgrade_log_item = OpenyUpgradeLog::create([
+      'user_id' => 1,
+      'name' => $config_name,
+      'created' => $logger_entity->getCreatedTime(),
+    ]);
+    $new_upgrade_log_item->setData($configFactory->get($config_name)->getOriginal());
+    $new_upgrade_log_item->setNewRevision(TRUE);
+    $new_upgrade_log_item->setRevisionUserId(1);
+    $new_upgrade_log_item->setRevisionCreationTime(time());
+    $new_upgrade_log_item->setRevisionLogMessage('Migrate from logger entity');
+    $new_upgrade_log_item->save();
+    $sandbox['progress']++;
+    $sandbox['current'] = $logger_entity->id();
+    $logger_entity->delete();
+  }
+
+  $sandbox['#finished'] = empty($sandbox['max']) ? 1 : ($sandbox['progress'] / $sandbox['max']);
+
+  if ($sandbox['#finished'] == 1) {
+    $entityTypeManager = \Drupal::entityTypeManager();
+    // Delete openy_config_upgrade_logs logger entity type.
+    $content_type = $entityTypeManager
+      ->getStorage('logger_entity_type')
+      ->load('openy_config_upgrade_logs');
+    $content_type->delete();
+    // No need to use old entity type and related module if other bundles
+    // not exist, so we can uninstall logger_entity module.
```

Comment:
> No need to use old entity type and related module if other bundles not exist

**if other bundles not exist** - after deleting `openy_config_upgrade_logs` bundle you can see this condition:
```
    $bundles = \Drupal::entityTypeManager()
      ->getStorage('logger_entity_type')
      ->loadMultiple();
    if (empty($bundles)) {
```

It will uninstall module only if other bundles not exist. So in case other custom module use `logger_entity` - it will leave module enabled.

I know that `logger_entity` used on several projects and in advance provided for this moment.

Reply:
I think we can mark this conversation as resolved.